hibernate-core/src/main/java/org/hibernate/metamodel/internal/AttributeFactory.java

```diff
@@ -228,9 +228,9 @@ public static <Y> DomainType<Y> determineSimpleType(ValueContext typeContext, Me
 					if ( domainType == null ) {
 						// Due to the use of generics, it can happen that a mapped super class uses a type
 						// for an attribute that is not a managed type. Since this case is not specifically mentioned
-						// in the Jakarta Persistence spec, we handle this by returning a "dummy" entity type
+						// in the Jakarta Persistence spec, we handle this by returning an Object entity type
 						final JavaType<Y> domainJavaType = context.getJavaTypeRegistry().resolveDescriptor(
-								typeContext.getJpaBindableType()
+								Object.class
 						);
```

Comment:
Doesn't this lead to reporting that the java type of the `parent` attribute is `Object` rather than the IMO more correct `AbstractTreeableEntity`?

Reply:
It does, but when the type is `AbstractTreeableEntity` queries on the `parent` attribute fail with the error as reported in the issue. This is because when executing queries the attribute `parent` is resolved to the mapper superclass, which is not a persistable Entity.

Reply:
@mbladel It introduce new problem: `root.get("parent").get("id")` will produce `org.hibernate.query.SemanticException: Could not resolve attribute 'id' of 'java.lang.Object'"`

Reply:
@quaff thank you for the feedback. This error is not caused by the edit, even if before it was more readable reporting `AbstractTreeableEntity` instead of `Object`.

Reply:
🤝 Just remind you this PR is not a complete fix.

Reply:
I think generic type should be resolved like this:
```java
Type resolvedType = org.springframework.core.GenericTypeResolver.resolveType(
		AbstractTreeableEntity.class.getDeclaredField("parent").getGenericType(), TestEntity.class);
assert (resolvedType == TestEntity.class);
```